simplesat/tests/test_solver.py

```diff
@@ -348,6 +351,87 @@ def test_missing_indirect_dependency_fails(self):
         with self.assertRaises(SatisfiabilityError):
             self.resolve(request)
 
+    def test_satisfy_requirements(self):
+        requirements = (
+            R(u'B ^= 1.0.0'),
+            R(u'E'),
+        )
+
+        packages = (
+            P(u"A 0.0.1-1;"),
+            P(u"A 1.0.0-1;"),
+            P(u"A 2.0.0-1;"),
+            P(u"A 2.1.0-1;"),
+            P(u"B 1.0.0-1; depends (A ^= 1.0, A != 2.1.0-1)"),
+            P(u"B 2.0.0-1; depends (A != 2.1.0-1)"),
+            P(u"C 1.0.0-1; depends (B > 2.0.0-1)"),
+            P(u"D 1.0.0-1;"),
+            P(u"E 1.0.0-1; depends (C > 2.0, D < 1.0)"),
+        )
+
+        modifiers = ConstraintModifiers(
+            allow_newer=[u'A'],
+            allow_older=[u'B'],
+            allow_any=[u'C', u'D'],
+        )
+
+        # When
+        result = satisfy_requirements(
+            packages, requirements, modifiers=modifiers)
+        expected = tuple(packages[i] for i in (2, 7, 4, 6, 8))
```

Comment:
The packages are in the order that the transaction has them, which, for installations, is  topological sorted.